app/models/processor.rb

```diff
@@ -0,0 +1,81 @@
+require 'json'
+require 'useragent'
+require 'uri'
+
+module Processor
```

Comment:
this module has some repetitive code. can we break these methods down so that we can reuse some of it?